### PR TITLE
GraduateNU V2 Infra - Docker and Staging CD

### LINKED ISSUE #479: Staging Infra
- Dockerize the frontend and backend by adding 2 docker files
- Enable CD so that pushing to main gets auto deployed
   - For testing purpose try changing it to deploying when you push to your branch. If it works change it to "on push to main".

### PR DESCRIPTION
**Pull request by aryanshah701**
# Description

## Changes
- Docker files to dockerize frontend and backend
- Uncomment and update CD workflow
- Add 4 new package.json scripts to build and run frontend and backend docker
- Add CI to test docker builds for frontend and backend
- Changes to AWS resources to make things work
- Minor frontend-v2 changes to fix the build after merging from main

## Noted issue
- Synchronize true in production. We should be using migrations. This is just staging and alpha release though, will definitely use migrations for beta production.

Closes #479 

# How Has This Been Tested?

- https://staging.graduatenu.com/
- running the new yarn commands: frontend:docker:build, frontend:docker:run, backend:docker:build, backend:docker:run

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code where needed 
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/frontend-v2/components/Sidebar/SectionRequirement.tsx [hunk 1]
```diff
@@ -127,6 +127,7 @@ const SectionRequirement: React.FC<SidebarRequirementProps> = ({
   const renderSection = (requirement: Section) => {
     return (
       <SidebarSection
+        isValid
```
Comment (aryanshah701) [inline packages/frontend-v2/components/Sidebar/SectionRequirement.tsx:130]: this prop was missing in main but required. @AlpacaFur @jia614, seems like we don't handle nested sections just yet, is this fine?

File: packages/frontend-v2/tsconfig.json [hunk 2]
```diff
@@ -15,7 +15,8 @@
     "jsx": "preserve",
     "incremental": true,
     "strictPropertyInitialization": false,
-    "experimentalDecorators": true
+    "experimentalDecorators": true,
+    "downlevelIteration": true
```
Comment (aryanshah701) [inline packages/frontend-v2/tsconfig.json:19]: needed to fix build after merging from main

File: packages/frontend-v2/components/Sidebar/SectionRequirement.tsx (hunk 1 above)
Comment (AlpacaFur) [reply packages/frontend-v2/components/Sidebar/SectionRequirement.tsx:130]: Yeah, that should be fine for now, we can take another look when we want to handle nested sections. (I'm not fully sure why the lint didn't catch the missing prop though 👀).

**Review by AlpacaFur — APPROVED** [review]
Looks good!

**Pull request merged**